standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java

```diff
@@ -0,0 +1,1641 @@
+      caseSensitive = false;
+    }
+
+    ConfVars(String varname, String hiveName, boolean defaultVal, String comment) {
+      this.varname = varname;
+      this.hiveName = hiveName;
+      this.defaultVal = defaultVal;
+      validator = null;
+      caseSensitive = false;
+    }
+
+    ConfVars(String varname, String hiveName, double defaultVal, String comment) {
+      this.varname = varname;
+      this.hiveName = hiveName;
+      this.defaultVal = defaultVal;
+      validator = null;
+      caseSensitive = false;
+    }
+
+    ConfVars(String varname, String hiveName, long defaultVal, TimeUnit unit, String comment) {
+      this.varname = varname;
+      this.hiveName = hiveName;
+      this.defaultVal = new TimeValue(defaultVal, unit);
+      validator = new Validator.TimeValidator(unit);
+      caseSensitive = false;
+    }
+
+    ConfVars(String varname, String hiveName, long defaultVal, TimeUnit unit,
+             Validator validator, String comment) {
+      this.varname = varname;
+      this.hiveName = hiveName;
+      this.defaultVal = new TimeValue(defaultVal, unit);
+      this.validator = validator;
+      caseSensitive = false;
+    }
+
+    public void validate(String value) throws IllegalArgumentException {
+      if (validator != null) {
+        validator.validate(value);
+      }
+    }
+
+    public boolean isCaseSensitive() {
+      return caseSensitive;
+    }
+
+    @Override
+    public String toString() {
+      return varname;
+    }
+  }
+
+  public static final ConfVars[] dataNucleusAndJdoConfs = {
+      ConfVars.AUTO_CREATE_ALL,
+      ConfVars.CONNECTION_DRIVER,
+      ConfVars.CONNECTION_POOLING_MAX_CONNECTIONS,
+      ConfVars.CONNECTION_POOLING_TYPE,
+      ConfVars.CONNECTURLKEY,
+      ConfVars.CONNECTION_USER_NAME,
+      ConfVars.DATANUCLEUS_AUTOSTART,
+      ConfVars.DATANUCLEUS_CACHE_LEVEL2,
+      ConfVars.DATANUCLEUS_CACHE_LEVEL2_TYPE,
+      ConfVars.DATANUCLEUS_INIT_COL_INFO,
+      ConfVars.DATANUCLEUS_PLUGIN_REGISTRY_BUNDLE_CHECK,
+      ConfVars.DATANUCLEUS_TRANSACTION_ISOLATION,
+      ConfVars.DATANUCLEUS_USE_LEGACY_VALUE_STRATEGY,
+      ConfVars.DETACH_ALL_ON_COMMIT,
+      ConfVars.IDENTIFIER_FACTORY,
+      ConfVars.MANAGER_FACTORY_CLASS,
+      ConfVars.MULTITHREADED,
+      ConfVars.NON_TRANSACTIONAL_READ,
+      ConfVars.PWD,
+      ConfVars.STORE_MANAGER_TYPE,
+      ConfVars.VALIDATE_COLUMNS,
+      ConfVars.VALIDATE_CONSTRAINTS,
+      ConfVars.VALIDATE_TABLES
+  };
+
+  // Make sure no one calls this
+  private MetastoreConf() {
+    throw new RuntimeException("You should never be creating one of these!");
+  }
+
+  public static Configuration newMetastoreConf() {
+
+    Configuration conf = new Configuration();
+
+    ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+    if (classLoader == null) {
+      classLoader = MetastoreConf.class.getClassLoader();
+    }
+    hiveDefaultURL = classLoader.getResource("hive-default.xml");
```

Comment:
Ok, makes sense.  I copied this functionality from HiveConf.  I think the reason it finds the file but doesn't add it is for getHiveDefaultLocation function, so that it knows where the file is.  But I should put a comment in to make that clear since I agree it's confusing.